tests/res_diff_expect/test_result_both_fail.ion

```diff
@@ -0,0 +1,20 @@
+$ion_1_0
+{
+ test: {
+  read_error: {
+   message: "Read_error: new commit causes new read error(s).",
+   read_error: {
+    new_errors: [
+     {
+      message: "test2."
+     }
+    ]
+   }
+  }
+ },
+ test2: {
+  result: {
+   message: "Result: two revisions have different results. This might be a cli tool issue."
```

Comment:
Similar to one of my previous questions, can you remind me if for `bad`, going from `FAIL` to `PASS` is good or bad?

I think either way, we should show here that the change occurred, but the `--results-diff` invocation should only return a failure code (thereby failing the Github Actions run) if the change was for the worse.